web/search.dart

```diff
@@ -251,21 +296,75 @@ void initializeSearch(
   }
 
   void showSuggestions() {
-    if (presentation.hasChildNodes()) {
+    if (searchResults.hasChildNodes()) {
       listBox.style.display = 'block';
       listBox.setAttribute('aria-expanded', 'true');
     }
   }
 
+  // Function that iterates through the categoriesMap and appends it to the given HTML element
+  // The received Element is the one used for search results page or dropdown list in the search bar
+  void iterateCategoriesMap(Element e) {
+    categoriesMap.forEach((keys, values) {
+      if (categoriesMap[keys] != null) {
```

Comment:
You don't have to do this check since you are holding onto the `values` value from line 308. I think just `e.append(values);`.